### PR TITLE
Changed: FetchHints to support more filtering

### PR DESCRIPTION
**Pull request by joeferner**

### CODE REVIEW

File: core/src/main/java/org/vertexium/mutation/ExistingElementMutationImpl.java [hunk 1]
```diff
@@ -76,7 +75,12 @@ public ExistingElementMutationImpl(T element) {
 
     @Override
     public ElementMutation<T> deleteProperty(Property property) {
-        FetchHint.checkFetchHints(getElement().getFetchHints(), EnumSet.of(FetchHint.PROPERTIES, FetchHint.PROPERTY_METADATA));
+        if (!element.getFetchHints().isIncludeAllProperties()) {
```
Comment (mwizeman) [inline core/src/main/java/org/vertexium/mutation/ExistingElementMutationImpl.java]: Same comment as above. Do we need to have loaded all of the properties to delete one? What if we loaded the one you are trying to delete?

File: core/src/main/java/org/vertexium/mutation/ExistingElementMutationImpl.java [hunk 2]
```diff
@@ -164,7 +168,12 @@ public ExistingElementMutationImpl(T element) {
 
     @Override
     public ExistingElementMutation<T> alterPropertyVisibility(Property property, Visibility visibility) {
-        FetchHint.checkFetchHints(getElement().getFetchHints(), EnumSet.of(FetchHint.PROPERTIES, FetchHint.PROPERTY_METADATA));
+        if (!element.getFetchHints().isIncludeAllProperties()) {
```
Comment (mwizeman) [inline core/src/main/java/org/vertexium/mutation/ExistingElementMutationImpl.java]: Same comment as above. Do we need all the properties or just the one you're altering?

File: core/src/main/java/org/vertexium/mutation/ExistingElementMutationImpl.java [hunk 3]
```diff
@@ -176,7 +185,12 @@ public ExistingElementMutationImpl(T element) {
 
     @Override
     public ExistingElementMutation<T> alterPropertyVisibility(String key, String name, Visibility visibility) {
-        FetchHint.checkFetchHints(getElement().getFetchHints(), EnumSet.of(FetchHint.PROPERTIES, FetchHint.PROPERTY_METADATA));
+        if (!element.getFetchHints().isIncludeAllProperties()) {
```
Comment (mwizeman) [inline core/src/main/java/org/vertexium/mutation/ExistingElementMutationImpl.java]: Same comment as above. Do we need all the properties or just the one you're altering?

File: accumulo-iterators/src/main/java/org/vertexium/accumulo/iterator/ElementIterator.java [hunk 4]
```diff
@@ -225,20 +250,72 @@ private void extractPropertyData(Key key, Value value) {
...
+            return;
+        }
+        iteratorSettings.addOption(key, value);
+    }
+
+    private Set<String> parseSet(String str) {
+        if (str == null) {
+            return null;
+        }
+        String[] parts = str.split("" + RECORD_SEPERATOR);
```
[the remark this answers is not in the document]
Comment (joeferner) [reply accumulo-iterators/src/main/java/org/vertexium/accumulo/iterator/ElementIterator.java]: switched to `Pattern.compile()` constant to be faster

File: accumulo-iterators/src/main/java/org/vertexium/accumulo/iterator/ElementIterator.java [hunk 5]
```diff
@@ -41,12 +41,13 @@
...
-    private static final String SETTING_FETCH_HINTS = "fetchHints";
-    private EnumSet<IteratorFetchHint> fetchHints;
+    private static final String SETTING_FETCH_HINTS_PREFIX = "fetchHints.";
+    private static final char RECORD_SEPERATOR = '\u001f';
+    private IteratorFetchHints fetchHints;
     private T elementData;
     private static final Map<Text, PropertyMetadataColumnQualifier> stringToPropertyMetadataColumnQualifierCache = new HashMap<>();
 
-    public ElementIterator(SortedKeyValueIterator<Key, Value> source, EnumSet<IteratorFetchHint> fetchHints) {
+    public ElementIterator(SortedKeyValueIterator<Key, Value> source, IteratorFetchHints fetchHints) {
```
[the remark this answers is not in the document]
Comment (joeferner) [reply accumulo-iterators/src/main/java/org/vertexium/accumulo/iterator/ElementIterator.java]: Looks like other users of this class do not call `init`, I think only Accumulo tablets call init when used.

File: accumulo/src/main/java/org/vertexium/accumulo/AccumuloGraph.java [hunk 6]
```diff
@@ -477,7 +478,12 @@ void saveProperties(
     }
 
     void deleteProperty(AccumuloElement element, Property property, Authorizations authorizations) {
-        FetchHint.checkFetchHints(element.getFetchHints(), EnumSet.of(FetchHint.PROPERTIES, FetchHint.PROPERTY_METADATA));
+        if (!element.getFetchHints().isIncludeAllProperties()) {
```
[the remark this answers is not in the document]
Comment (joeferner) [reply accumulo/src/main/java/org/vertexium/accumulo/AccumuloGraph.java]: done, now checking property name is included along with all metadata

File: core/src/main/java/org/vertexium/mutation/ExistingElementMutationImpl.java (hunk 1 above)
Comment (joeferner) [reply core/src/main/java/org/vertexium/mutation/ExistingElementMutationImpl.java]: done

File: core/src/main/java/org/vertexium/mutation/ExistingElementMutationImpl.java (hunk 2 above)
Comment (joeferner) [reply core/src/main/java/org/vertexium/mutation/ExistingElementMutationImpl.java]: done

File: core/src/main/java/org/vertexium/mutation/ExistingElementMutationImpl.java (hunk 3 above)
Comment (joeferner) [reply core/src/main/java/org/vertexium/mutation/ExistingElementMutationImpl.java]: done

**Pull request merged**